mrmustard/lab_dev/states/base.py

```diff
@@ -648,6 +631,29 @@ def __init__(
         if representation is not None:
             self._representation = representation
 
+    def auto_shape(self) -> tuple[int, ...]:
```

Comment:
I was surprised to see that this modifies the fock_shape of this object. There seem to be a lot of side effects throughout MrMustard, and I'm worried that they'll be hard for users to keep track of. Not for this PR, but I think it would be worth spending some time documenting (somewhere) what they all are, particularly in lab_dev before switching it over to lab.

Reply:
Yes, sounds good let's do that and think whether we need them. This specific side effect (which is documented in both places) does not modify the entries of `fock_shape` that are already set: it only completes what a user doesn't yet know (and in any case `fock_shape` is designed to be editable, so a user can still change it in the future).